src/components/DataSearch.jsx

```diff
@@ -0,0 +1,380 @@
+import React, { Component } from 'react';
+import {
+  bool,
+  string,
+  object,
+  number,
+  func,
+  dataField,
+  position,
+  suggestions,
+  highlightField,
+  queryFormat,
+  fuzziness,
+  title,
+  any,
+  react,
+  themePreset
+} from '../utils/types';
+import Input, {
+  suggestionsContainer,
+  suggestions as suggestionsCss
+} from './Input';
+import Container from '../styles/Container';
+import Title from '../styles/Title';
+import {
+  deepGet,
+  getClassName,
+  getComponent,
+  hasCustomRenderer
+} from '../utils/helper';
+import Downshift from 'downshift';
+import Icons from './Icons';
+import Loader from './Loader';
+import Error from './Error';
+import SuggestionItem from '../addons/SuggestionItem';
+import NoSuggestions from './NoSuggestions';
+import Searchbase from '@appbaseio/searchbase';
+
+class DataSearch extends Component {
+  constructor(props) {
+    super(props);
+    const { index, url, dataField, credentials } = props;
+    const currentValue = props.value || props.defaultValue || '';
+
+    this.state = {
+      currentValue,
+      suggestionsList: [],
+      isOpen: false
+    };
+
+    this.searchBase = new Searchbase({
+      index,
+      url,
+      dataField,
+      credentials
+    });
+
+    this.searchBase.subscribeToStateChanges(
```

Comment:
How about using `onSuggestion` instead ?

Reply:
`onSuggestion(prev, next)` => here also `prev` and `next` are getting the same value

Reply:
Oh, not about the data. I am asking what if we use `onSuggestion` instead of `subscribe` as the library would only focus on Suggestions.

Reply:
> Oh, not about the data. I am asking what if we use `onSuggestion` instead of `subscribe` as the library would only focus on Suggestions.

`state management` have been done in the subscribe method and callbacks have been handled in this way, just to keep both of them independent
```js
this.searchBase.onSuggestion = props.onSuggestion
this.searchBase.onError = props.onError
```
If I had handled state in onSuggestion, then would have to handle two conditions in one function,
```js
this.searchBase.onSuggestion = (prev, next) => {
    setState({});
    if (props.onSuggestions) props.onSuggestion(prev, next)
}
```
Any specific rationale behind handling state in onSuggestion?

Reply:
Subscribe would be called on every state changes like result, pending state and other state changes but onSuggestion is only called when we get suggestions. So we render only when we get suggestion.

Reply:
no, we can define on which properties to subscribe
![image](https://user-images.githubusercontent.com/10362927/65676960-6494ab80-e06e-11e9-94ae-ccdc5e4f5b4c.png)

Reply:
Cool. Do unsubscribe it on unmount